### ISSUE TITLE
"kubectl": executable file not found in $PATH

### ISSUE THREAD

**Issue by konstantinpae** [loft-sh/devpod#414]
**What happened?**
kubectl not found

**What did you expect to happen instead?**
kubectl to be found as it is on the $PATH (in bash as well zsh)

```
REDACTED@REDACTED  ~  which kubectl
/Users/REDACTED/.rd/bin/kubectl
```

```
bash-3.2$ which kubectl
/Users/REDACTED/.rd/bin/kubectl
```

```
REDACTED@REDACTED  ~  echo $PATH
/Users/REDACTED/.krew/bin:/Users/REDACTED/Tools:/Users/REDACTED/.poetry/bin:/Users/REDACTED/.rd/bin:/Users/REDACTED/bin:/usr/local/bin:/opt/homebrew/bin:/usr/local/bin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/homebrew/bin:/Library/Apple/usr/bin:/Applications/Wireshark.app/Contents/MacOS
```

My `devcontainer.json`:

```
{
    "name": "REDACTED",
    "build": {
        "dockerfile": "Dockerfile",
        "context": ".."
    },
    "customizations": {
        "vscode": {
            "extensions": [
                "ms-kubernetes-tools.vscode-kubernetes-tools",
                "ms-azuretools.vscode-docker",
                "golang.go",
                "Vue.volar",
                "Vue.vscode-typescript-vue-plugin"
            ]
        }
    }
}
```

Debug output:

```
[12:59:09] debug Execute command locally
[12:59:09] debug Use /Users/REDACTED.devpod/agent/contexts/default/workspaces/REDACTED as workspace dir
[12:59:09] debug Delete old workspace 'REDACTED'
[12:59:09] debug Removing DevPod container from server...
[12:59:09] debug Use Kubernetes Namespace 'devpod'
[12:59:09] debug Use Kubernetes Config '/Users/REDACTED.kube/config'
[12:59:09] debug Use Kubernetes Context 'rancher-desktop'
[12:59:09] debug Run command: kubectl --namespace devpod --kubeconfig /Users/REDACTED.kube/config --context rancher-desktop get pvc devpod-REDACTED --ignore-not-found -o json
[12:59:09] debug error parsing workspace info: delete old workspace: remove container: find dev container: find pvc: exec: "kubectl": executable file not found in $PATH
[12:59:09] debug Done executing up command
[12:59:09] debug Done creating devcontainer
```

**Local Environment:**

* DevPod Version: 0.1.9
* Operating System:  MacOS Ventura 13.2.1
* ARCH of the OS: ARM64

**DevPod Provider:**

* Kubernetes Provider: Client Version: [version.Info](http://version.Info){Major:"1", Minor:"23", GitVersion:"v1.23.3", GitCommit:"816c97ab8cff8a1c72eccca1026f7820e93e0d25", GitTreeState:"clean", BuildDate:"2022-01-25T21:25:17Z", GoVersion:"go1.17.6", Compiler:"gc", Platform:"darwin/amd64"}
  Server Version: [version.Info](http://version.Info){Major:"1", Minor:"24", GitVersion:"v1.24.3+k3s1", GitCommit:"990ba0e88c90f8ed8b50e0ccd375937b841b176e", GitTreeState:"clean", BuildDate:"2022-07-19T01:08:03Z", GoVersion:"go1.18.1", Compiler:"gc", Platform:"linux/arm64"}
* Local/remote provider: docker

**Anything else we need to know?**
Not sure if this is a bug at all, or if I need to set the PATH in a different context. If not a bug, please feel free to close.

**Comment by FabianKramm** [comment]
@konstantinpae thanks for creating this issue! How do you set this to the path? Is it in `~/.zshrc`?

**Comment by konstantinpae** [comment]
@FabianKramm:  The path to kubectl is set in $PATH in .zshrc as well as .bashrc:

.zshrc
```
...
### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)
export PATH="/Users/REDACTED/.rd/bin:$PATH"
```
.bashrc
```
bash-3.2$ cat .bashrc
### MANAGED BY RANCHER DESKTOP START (DO NOT EDIT)
export PATH="/Users/REDACTED.rd/bin:$PATH"
```
As advised I also set the path to kubectl directly in the UI
<img width="323" alt="image" src="https://github.com/loft-sh/devpod/assets/7922974/3d8099fe-6e50-449f-bfc8-e6e887315646">

Additionally, I completely started from scratch, means wiping all existing providers and workspaces.
The same error still is showing up:
<img width="965" alt="image" src="https://github.com/loft-sh/devpod/assets/7922974/78db65bf-da5a-4a01-b9b7-230b496c2eba">
Debug Logs:
```
[08:39:09] debug Execute command locally
[08:39:09] debug Use /Users/REDACTED/.devpod/agent/contexts/default/workspaces/omega as workspace dir
[08:39:09] debug Delete old workspace 'REDACTED'
[08:39:09] debug Removing DevPod container from server...
[08:39:09] debug Use Kubernetes Namespace 'devpod'
[08:39:09] debug Use Kubernetes Config '/Users/REDACTED/.kube/config'
[08:39:09] debug Use Kubernetes Context 'rancher-desktop'
[08:39:09] debug Run command: kubectl --namespace devpod --kubeconfig /Users/REDACTED/.kube/config --context rancher-desktop get pvc devpod-REDACTED --ignore-not-found -o json
[08:39:09] debug error parsing workspace info: delete old workspace: remove container: find dev container: find pvc: exec: "kubectl": executable file not found in $PATH
[08:39:09] debug Done executing up command
[08:39:09] debug Done creating devcontainer
```

**Comment by FabianKramm** [comment]
@konstantinpae mhh strange I tried to reproduce this on my machine with installing rancher desktop and deleting all other kubectl binaries and its still working fine for me. 

When does this error come up exactly? Is DevPod able to create a new workspace and then this shows up when deleting the workspace? It also seems you came past the provider init step, which should check if kubectl can be executed, so there might be something else that might cause issues there.

**Comment by konstantinpae** [comment]
@FabianKramm: 
Seems I got it working, after testing on a different machine without any problems, I updated my k8s client/server to stable (1.26.5) which I did not have because of an older project. Then I completely wiped the cluster and set it up anew. 

So I suspect there was something wrong with the versions of kubectl. 

As a side note, somehow an authenticated pull of an image is not possible. I'm using dockerd as container engine instead of containerd which in turn relies on docker-credential-osxkeychain which then is not found on the path (same error message as above, made sure it is on path the same way as above). 

In conclusion, I got it running now if somehow shaky. Also switching from dockerd to containerd seems to improve stability, at least when running on Rancher Desktop.

As regarding your question, the error came up exclusively when creating a workspace (the provider init worked flawless in this case, I had it failing once because of a typo).

Thank you for your help and please feel free to close this issue.
